In [1]:
import pandas as pd
from pathlib import Path

In [52]:
input_dir = Path("../../output_nuclear/all/")
csv_files = list(input_dir.glob("*.csv"))
print(csv_files)
dfs = []
for fp in csv_files:
    df = pd.read_csv(fp)
    df = df[["file_path", "date", "article_id", "title", "body", "label"]]
    dfs.append(df)

combined = pd.concat(dfs, ignore_index=True)

#combined.to_csv(input_dir / "combined_all.csv", index=False, encoding="utf-8")


[PosixPath('../../output_nuclear/all/to_label_articles_1960.csv'), PosixPath('../../output_nuclear/all/to_label_articles_1970.csv'), PosixPath('../../output_nuclear/all/to_label_articles_1960_ddd.csv'), PosixPath('../../output_nuclear/all/to_label_articles_1970_ddd.csv'), PosixPath('../../output_nuclear/all/to_label_articles_1980_ddd.csv'), PosixPath('../../output_nuclear/all/to_label_articles_1990_ddd.csv'), PosixPath('../../output_nuclear/all/to_label_articles_1980.csv'), PosixPath('../../output_nuclear/all/to_label_articles_1990.csv')]


In [53]:
combined['body_length'] = combined['body'].str.len()
combined['word_count'] = combined['body'].str.split().apply(len)

In [54]:
combined.shape

(43595, 8)

In [55]:
combined

file_path        date  \
0      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1966-06-27   
1      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
2      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
3      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
4      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
...                                                  ...         ...   
43590  ../data/converted/2022_harvest_KRANTEN/20/KRAN...  1994-11-23   
43591  ../data/converted/2022_harvest_KRANTEN/27/KRAN...  1994-03-31   
43592  ../data/converted/2022_harvest_KRANTEN/27/KRAN...  1994-03-31   
43593  ../data/converted/2022_harvest_KRANTEN/27/KRAN...  1994-03-31   
43594  ../data/converted/2022_harvest_KRANTEN/27/KRAN...  1994-03-31   

       article_id                                             title  \
0              45           Produktie kernenergie stijgt razendsnel   
1             131  In Nederland voorlopig geen kernenergiecentrales   
2             131  In Nederland voorlopig geen kernenergiecentrales   
3             131  In Nederland voorlopig geen kernenergiecentrales   
4             131  In Nederland voorlopig geen kernenergiecentrales   
...           ...                                               ...   
43590           2                          Toch besluit kernenergie   
43591          24                       NUCLEAIRE TIJDBOM TIKT DOOR   
43592          24                       NUCLEAIRE TIJDBOM TIKT DOOR   
43593          24                       NUCLEAIRE TIJDBOM TIKT DOOR   
43594          24                       NUCLEAIRE TIJDBOM TIKT DOOR   

                                                    body  label  body_length  \
0      BOEDAPEST (ÜPI) — Omstreeks l»»u zullen kernce...    NaN          736   
1      Prof. Van Lier aanvaardt ambt aan t.h.-Delft P...    NaN         1178   
2      meer het geval zijn. De kernenergie- ?P centra...    NaN         3254   
3      Minister Marijnen niet te vermurwen (Van onze ...    NaN         1503   
4      Centrale training Ned. elftal begint 9 maart D...    NaN         1574   
...                                                  ...    ...          ...   
43590  sociaal-democraten Van Heemst en Woltjer zitte...    NaN          657   
43591  door PETER D’HAMECOURT MOSKOU — Een nucleaire ...    NaN         2678   
43592  de burelen van de Moskouse ministeries die bel...    NaN          809   
43593  groep van experts van het milieuministerie de ...    NaN          842   
43594  experts geloven niet dat de zee rond de ‘Komso...    NaN          762   

       word_count  
0             110  
1             184  
2             551  
3             222  
4             238  
...           ...  
43590          99  
43591         423  
43592         115  
43593         134  
43594         125  

[43595 rows x 8 columns]

In [56]:
lengths = combined["body"].astype(str).str.len()

min_len = combined['body_length'].min()
max_len = combined['body_length'].max()
avg_len = combined['body_length'].mean()

print(f"Body length — min: {min_len}, max: {max_len}, avg: {avg_len:.1f}")


Body length — min: 200, max: 31664, avg: 643.1


In [57]:
lengths = combined["body"].astype(str).str.len()

min_len = combined['word_count'].min()
max_len = combined['word_count'].max()
avg_len = combined['word_count'].mean()

print(f"Body words — min: {min_len}, max: {max_len}, avg: {avg_len:.1f}")


Body words — min: 22, max: 5079, avg: 98.7


In [58]:
combined[combined['word_count']<340]

file_path        date  \
0      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1966-06-27   
1      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
3      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
4      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
5      ../data/converted/2022_harvest_KRANTEN/47/KRAN...  1960-02-25   
...                                                  ...         ...   
43589  ../data/converted/2022_harvest_KRANTEN/20/KRAN...  1994-11-23   
43590  ../data/converted/2022_harvest_KRANTEN/20/KRAN...  1994-11-23   
43592  ../data/converted/2022_harvest_KRANTEN/27/KRAN...  1994-03-31   
43593  ../data/converted/2022_harvest_KRANTEN/27/KRAN...  1994-03-31   
43594  ../data/converted/2022_harvest_KRANTEN/27/KRAN...  1994-03-31   

       article_id                                             title  \
0              45           Produktie kernenergie stijgt razendsnel   
1             131  In Nederland voorlopig geen kernenergiecentrales   
3             131  In Nederland voorlopig geen kernenergiecentrales   
4             131  In Nederland voorlopig geen kernenergiecentrales   
5             131  In Nederland voorlopig geen kernenergiecentrales   
...           ...                                               ...   
43589           2                          Toch besluit kernenergie   
43590           2                          Toch besluit kernenergie   
43592          24                       NUCLEAIRE TIJDBOM TIKT DOOR   
43593          24                       NUCLEAIRE TIJDBOM TIKT DOOR   
43594          24                       NUCLEAIRE TIJDBOM TIKT DOOR   

                                                    body  label  body_length  \
0      BOEDAPEST (ÜPI) — Omstreeks l»»u zullen kernce...    NaN          736   
1      Prof. Van Lier aanvaardt ambt aan t.h.-Delft P...    NaN         1178   
3      Minister Marijnen niet te vermurwen (Van onze ...    NaN         1503   
4      Centrale training Ned. elftal begint 9 maart D...    NaN         1574   
5      Gedaagde tekenaar van • naakt kreeg vrijspraak...    NaN         1624   
...                                                  ...    ...          ...   
43589  VERVOLG VAN PAGINA 1 De stemming van vandaag i...    NaN          591   
43590  sociaal-democraten Van Heemst en Woltjer zitte...    NaN          657   
43592  de burelen van de Moskouse ministeries die bel...    NaN          809   
43593  groep van experts van het milieuministerie de ...    NaN          842   
43594  experts geloven niet dat de zee rond de ‘Komso...    NaN          762   

       word_count  
0             110  
1             184  
3             222  
4             238  
5             268  
...           ...  
43589         102  
43590          99  
43592         115  
43593         134  
43594         125  

[42475 rows x 8 columns]

In [59]:
df_sample = combined[combined['word_count']<340].sample(n=10000, random_state=42)  # random_state for reproducibility
df_sample['date'] = pd.to_datetime(df_sample['date'], errors='coerce')
df_sample['decade'] = (df_sample['date'].dt.year// 10) *10
df_sample['energy_resource']= 'nuclear energy'
df_sample.rename(columns={'body': 'text'}, inplace=True)


In [60]:
df_sample.to_csv(input_dir / "sample_340words.csv", index=False, encoding="utf-8")

In [61]:
df_sample['word_count'].mean()

88.362

So let's say 115 tokens in average for 88 words

In [19]:
df_sample.head()

file_path        date  \
10179  ../data/converted/2022_harvest_DDD/05/DDD_0109...  1977-04-07   
34224  ../data/converted/2022_harvest_DDD/80/DDD_0105...  1980-04-29   
702    ../data/converted/2022_harvest_KRANTEN/12/KRAN...  1979-10-13   
4463   ../data/converted/2022_harvest_DDD/94/DDD_0112...  1963-08-31   
12477  ../data/converted/2022_harvest_DDD/66/DDD_0110...  1979-10-30   

       article_id                                              title  \
10179         259  Eerst akkoord over voorwaarden Canada streng b...   
34224          64                                        Kernenergie   
702            42                         Drenthe tegen kerncentrale   
4463          403                 Kernenergie nog vóór 1970 rendabel   
12477         186  Centrale van ruim miljard staat ongebruikt Oos...   

                                                    body  label  body_length  \
10179  v OTTAWA' - Canada"-lis voorlopig '.nog niet v...    NaN         1219   
34224  De algemene vergadering aanvaardde ook een mot...    NaN          249   
702    ASSEN - Gedeputeerde staten van Drenthe hebben...    NaN          285   
4463   Van onze redacteur wetenschappen AMSTERDAM, za...    NaN          455   
12477  De in de oppositie zijnde conservatieve Oosten...    NaN          434   

       word_count  
10179         184  
34224          39  
702            49  
4463           67  
12477          65

In [20]:
# Convert to datetime format
df_sample['date'] = pd.to_datetime(df_sample['date'])

# Create a new column for the decade
df_sample['decade'] = (df_sample['date'].dt.year // 10) * 10

# Count rows per decade
decade_counts = df_sample['decade'].value_counts().sort_index()

print(decade_counts)

decade
1960    1504
1970    3105
1980    4193
1990    1198
Name: count, dtype: int64


In [21]:
combined['date'] = pd.to_datetime(combined['date'])

# Create a new column for the decade
combined['decade'] = (combined['date'].dt.year // 10) * 10

# Count rows per decade
decade_counts = combined['decade'].value_counts().sort_index()

print(decade_counts)

decade
1960     6358
1970    13604
1980    18556
1990     5077
Name: count, dtype: int64


## Renewable energy

In [62]:
input_dir = Path("../../output_renewable/all/")
csv_files = list(input_dir.glob("*.csv"))
print(csv_files)
dfs = []
for fp in csv_files:
    df = pd.read_csv(fp)
    df = df[["file_path", "date", "article_id", "title", "body", "label"]]
    dfs.append(df)

combined = pd.concat(dfs, ignore_index=True)

#combined.to_csv(input_dir / "combined_all.csv", index=False, encoding="utf-8")


[PosixPath('../../output_renewable/all/to_label_articles_1960.csv'), PosixPath('../../output_renewable/all/to_label_articles_1970.csv'), PosixPath('../../output_renewable/all/to_label_articles_1960_ddd.csv'), PosixPath('../../output_renewable/all/to_label_articles_1970_ddd.csv'), PosixPath('../../output_renewable/all/to_label_articles_1980_ddd.csv'), PosixPath('../../output_renewable/all/to_label_articles_1990_ddd.csv'), PosixPath('../../output_renewable/all/to_label_articles_1980.csv'), PosixPath('../../output_renewable/all/to_label_articles_1990.csv')]


In [63]:
combined['body_length'] = combined['body'].str.len()
combined['word_count'] = combined['body'].str.split().apply(len)

In [64]:
combined.shape

(7389, 8)

In [65]:
lengths = combined["body"].astype(str).str.len()

min_len = combined['body_length'].min()
max_len = combined['body_length'].max()
avg_len = combined['body_length'].mean()

print(f"Body length — min: {min_len}, max: {max_len}, avg: {avg_len:.1f}")

Body length — min: 200, max: 16974, avg: 672.9


In [66]:
lengths = combined["body"].astype(str).str.len()

min_len = combined['word_count'].min()
max_len = combined['word_count'].max()
avg_len = combined['word_count'].mean()

print(f"Body words — min: {min_len}, max: {max_len}, avg: {avg_len:.1f}")

Body words — min: 24, max: 2632, avg: 105.3


In [67]:
combined.loc[combined['word_count']>=340,]

file_path        date  \
0     ../data-shiva/converted/2022_harvest_KRANTEN/0...  1966-10-04   
34    ../data-shiva/converted/2022_harvest_KRANTEN/2...  1961-01-26   
37    ../data-shiva/converted/2022_harvest_KRANTEN/2...  1963-07-27   
48    ../data-shiva/converted/2022_harvest_KRANTEN/2...  1968-01-24   
54    ../data-shiva/converted/2022_harvest_KRANTEN/1...  1963-01-12   
...                                                 ...         ...   
7327  ../data-shiva/converted/2022_harvest_KRANTEN/2...  1990-10-23   
7338  ../data-shiva/converted/2022_harvest_KRANTEN/2...  1992-04-24   
7354  ../data-shiva/converted/2022_harvest_KRANTEN/1...  1990-11-15   
7358  ../data-shiva/converted/2022_harvest_KRANTEN/1...  1993-06-14   
7377  ../data-shiva/converted/2022_harvest_KRANTEN/1...  1994-12-17   

      article_id                                             title  \
0            115            GROOTS PERSPECTIEF VOOR STIRLING MOTOR   
34           238  INDIË VERBAASDE VER. STATEN MET WATERSTOFENERGIE   
37           194                                 OPERATIE - SYNCOM   
48            96  Amerika zoekt naar vier verloren waterstofbommen   
54           141                FRANSEN GAAN WATERSTOFBOM PROBEREN   
...          ...                                               ...   
7327          29                     Statiegeld is zo gek nog niet   
7338         140                      GUADALAJARA ZOEKT SCHULDIGEN   
7354          96                                          HET WEER   
7358          33                                          HET WEER   
7377         245                                           Vandaag   

                                                   body  label  body_length  \
0     OXFORD — Voortgang makend op een zeer breed fr...    NaN         2926   
34    Indrukwekkende reuzensprong Van onze redacteur...    NaN         2990   
37    Zodra de Syncom 2 zijn bijna sta tionaire plaa...    NaN         3434   
48    WASHINGTON (Reuter I — J|et hondesleden, fakke...    NaN         2498   
54    PROBEREN i- Volgens kringen, die in nauw '0 Co...    NaN         2328   
...                                                 ...    ...          ...   
7327  auDieus is, oi de sloper bij die zeldzame gele...    NaN         2969   
7338  „Guadalajara wil dat de schuldigen worden bere...    NaN         2401   
7354  Weerrapporten a m m n van gisteravond 20 uur c...    NaN         1642   
7358  neerslag van 20 tot 20 uur S 3 n.o. = niet ont...    NaN         1542   
7377  gisteren weer wind m/s ïau neersi • 0° wzw 3 5...    NaN         2823   

      word_count  
0            415  
34           454  
37           561  
48           387  
54           414  
...          ...  
7327         438  
7338         361  
7354         416  
7358         396  
7377         714  

[243 rows x 8 columns]

In [68]:
df_sample = combined[combined['word_count']<340]  # random_state for reproducibility
df_sample['date'] = pd.to_datetime(df_sample['date'], errors='coerce')
df_sample['decade'] = (df_sample['date'].dt.year// 10) *10
df_sample['energy_resource']= 'renewable energy'
df_sample.rename(columns={'body': 'text'}, inplace=True)

/var/folders/1_/p59mzyv14zvb16j4tkwx37m00000gp/T/ipykernel_53540/3283601024.py:2: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_sample['date'] = pd.to_datetime(df_sample['date'], errors='coerce')
/var/folders/1_/p59mzyv14zvb16j4tkwx37m00000gp/T/ipykernel_53540/3283601024.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  df_sample['decade'] = (df_sample['date'].dt.year// 10) *10
/var/folders/1_/p59mzyv14zvb16j4tkwx37m00000gp/T/ipykernel_53540/3283601024.py:4: SettingWithCopyWarning: 
A value is trying t

In [69]:
df_sample.to_csv(input_dir / "sample_340words.csv", index=False, encoding="utf-8")

In [70]:
df_sample.shape

(7146, 10)

In [71]:
df_sample['word_count'].mean()

91.42975090959978

In [74]:
df_sample.head().to_csv(input_dir / "sample_head.csv", index=False, encoding="utf-8")

## make train-test dataset

In [ ]:
n_fp = "../../output_nuclear/all/sentiments_sonnet.csv"
df_n = pd.read_csv(n_fp)
df_n['label'] = df_n['label_ai']
df_n['label']= df_n['label'].replace(-1,3)
df_n = df_n[df_n['label'].notnull()]

In [ ]:
from sklearn.model_selection import train_test_split

# Assume df is your DataFrame
train_df, test_df = train_test_split(df_n, test_size=0.2, random_state=42)
train_df.to_csv("../../data/nuclear/train.csv", index = False)
test_df.to_csv("../../data/nuclear/test.csv", index = False)

In [ ]:
train_df['label']= train_df['label'].replace(2,1)
train_df.to_csv("../../data/nuclear/train_binary.csv", index = False)
test_df['label']= test_df['label'].replace(2,1)
test_df.to_csv("../../data/nuclear/test_binary.csv", index = False)

In [ ]:
r_fp = "../../output_renewable/all/sentiments_sonnet.csv"
df_r = pd.read_csv(r_fp)
df_r['label'] = df_r['label_ai']
df_n['label']= df_n['label'].replace(-1,3)
df_n = df_n[df_n['label'].notnull()]

In [ ]:
train_df, test_df = train_test_split(df_r, test_size=0.2, random_state=42)
train_df.to_csv("../../data/renewable/train.csv", index = False)
test_df.to_csv("../../data/renewable/test.csv", index = False)

In [ ]:
train_df['label']= train_df['label'].replace(2,1)
train_df.to_csv("../../data/renewable/train_binary.csv", index = False)
test_df['label']= test_df['label'].replace(2,1)
test_df.to_csv("../../data/renewable/test_binary.csv", index = False)